File: server/src/main/java/com/edutech/progressive/entity/Clinic.java
package com.edutech.progressive.entity;

import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;

import com.fasterxml.jackson.annotation.JsonIgnore;

@Entity
public class Clinic {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int clinicId;

    private String clinicName;
    private String location;

    private int doctorId; 

    private String contactNumber;
    private int establishedYear;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "doctorId", referencedColumnName = "doctorId", insertable = false, updatable = false)
    @JsonIgnore 
    private Doctor doctor;


    public Clinic(int clinicId, String clinicName, String location, int doctorId, String contactNumber, int establishedYear, Doctor doctor) {
        this.clinicId = clinicId;
        this.clinicName = clinicName;
        this.location = location;
        this.doctorId = doctorId;
        this.contactNumber = contactNumber;
        this.establishedYear = establishedYear;
        this.doctor = doctor;
    }

    public Clinic(){}

    public Clinic(int clinicId, String clinicName, String location, int doctorId, String contactNumber, int establishedYear) {
        this.clinicId = clinicId;
        this.clinicName = clinicName;
        this.location = location;
        this.doctorId = doctorId;
        this.contactNumber = contactNumber;
        this.establishedYear = establishedYear;
    }

    public int getClinicId() {
        return clinicId;
    }

    public void setClinicId(int clinicId) {
        this.clinicId = clinicId;
    }

    public String getClinicName() {
        return clinicName;
    }

    public void setClinicName(String clinicName) {
        this.clinicName = clinicName;
    }

    public String getLocation() {
        return location;
    }

    public void setLocation(String location) {
        this.location = location;
    }

    public int getDoctorId() {
        return doctorId;
    }

    public void setDoctorId(int doctorId) {
        this.doctorId = doctorId;
    }

    public String getContactNumber() {
        return contactNumber;
    }

    public void setContactNumber(String contactNumber) {
        this.contactNumber = contactNumber;
    }

    public int getEstablishedYear() {
        return establishedYear;
    }

    public void setEstablishedYear(int establishedYear) {
        this.establishedYear = establishedYear;
    }

    public Doctor getDoctor() {
        return doctor;
    }

    public void setDoctor(Doctor doctor) {
        this.doctor = doctor;
    }

    

    
    
    
}

File: client/src/app/mediconnect/models/Clinic.ts
import { Doctor } from "./Doctor";
export class Clinic {
    clinicId: number;
    clinicName: string;
    location: string;
    doctor: Doctor;
    contactNumber: string;
    establishedYear: number;

    constructor(clinicId: number,clinicName: string,location: string,doctor: Doctor,contactNumber: string,establishedYear: number)
    {
        this.clinicId=clinicId;
        this.clinicName=clinicName;
        this.location=location;
        this.doctor=doctor;
        this.contactNumber=contactNumber;
        this.establishedYear=establishedYear;
    }

    // logAttributes():void{
    //     console.log("clinicId:",this.clinicId);
    //     console.log("clinicName:",this.clinicName);
    //     console.log("location:",this.location);
    //     console.log("doctorId:",this.doctor);
    //     console.log("contactNumber:",this.contactNumber);
    //     console.log("establishedYear:",this.establishedYear);
        
        
        
        
        
        
    // }

}